sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/ResourceGroupOperations.cs

```diff
@@ -367,11 +373,10 @@ public virtual ResourceGroupCreateOrUpdateOperation StartAddTag(string key, stri
 
             try
             {
-                var resource = GetResource(cancellationToken);
-                var patch = new ResourceGroupPatchable();
-                patch.Tags.ReplaceWith(resource.Data.Tags);
-                patch.Tags[key] = value;
-                var originalResponse = RestClient.Update(Id.Name, patch, cancellationToken);
+                var originalTags = TagsRestClient.GetAtScope(Id, cancellationToken).Value;
```

Comment:
Using the tags rest client won't scale outside of core.  We need to write this in a way that will work both inside the core library and outside.

Reply:
Done